packages/components/src/components/Log/Log.js

```diff
@@ -92,27 +92,50 @@ export class LogContainer extends Component {
     }
   };
 
-  /* istanbul ignore next */
-  initPolling = () => {
-    const { stepStatus, pollingInterval } = this.props;
-    if (!this.timer && stepStatus && !stepStatus.terminated) {
-      this.timer = setInterval(() => this.loadLog(), pollingInterval);
+  readChunks = ({ done, value }, decoder, text = '') => {
+    if (done) {
+      return undefined;
     }
-    if (this.timer && stepStatus && stepStatus.terminated) {
-      clearInterval(this.timer);
+    if (this.canceled) {
+      this.reader.cancel();
+      return undefined;
     }
+    const logs =
+      text + decoder.decode(value || new Uint8Array(), { stream: !done });
+    this.setState({
+      loading: false,
+      logs: logs.split('\n')
+    });
+    return this.reader
+      .read()
+      .then(result => this.readChunks(result, decoder, logs));
   };
 
   loadLog = async () => {
-    const { fetchLogs, intl } = this.props;
+    const { fetchLogs, intl, stepStatus, pollingInterval } = this.props;
     if (fetchLogs) {
       try {
         const logs = await fetchLogs();
-        this.setState({
-          loading: false,
-          logs: logs ? logs.split('\n') : undefined
-        });
-      } catch {
+        if (logs instanceof ReadableStream) {
+          const decoder = new TextDecoder();
+          this.reader = logs.getReader();
+          this.reader
+            .read()
+            .then(result => this.readChunks(result, decoder))
+            .catch(error => {
+              throw error;
+            });
```

Comment:
this error isn't being handled
if you want it to be caught by the `catch` block below you'll need to do something like `await this.reader.read()...`

Reply:
Good catch !

Reply:
Fixed